Import the `wildargs` library: automatic expansion of wildcard arguments

ctruta opened the pull request:
Let this be _"it ain't stolen if it's from your own project"_ kind of contribution.

I've been using this micro-project, wildly named "wildargs", in OptiPNG, for about 22 years (according to the change log).

This micro-project (or micro-library) used to be licensed under the zlib license, like OptiPNG itself. But then, earlier this year, I changed the license to Boost, which is one of the most liberal licenses out there. This article nails it:

_"Why You Should Use the Boost Software License"_ by Peter Dimov
https://pdimov.github.io/blog/2020/09/06/why-use-the-boost-license/

However: @svgeesus: if you do accept this wildargs thingy, but you would rather prefer to take it under a different license (e.g. pngcheck's own license), please tell me so, and I'll be happy to relicense.

Same question goes about your preference for a different subdirectory instead of `third_party`. If you want to have a look at how OptiPNG is set up, you can see it here: https://sourceforge.net/p/optipng/code/ci/tmp/main/~/tree/

ctruta commented:
FYI, I just forced-pushed changes to my proposed changes, because I had forgotten to make the updates to the change log. I claim that my changes, including the change log, are production-ready... which reminds me... we have yet to set up this project's CI. (IIRC, I actually promised I'll do it.)

ctruta commented:
I just wrote this in #45:

> In any case, @svgeesus, now you have two competing solutions to one problem. Plus one extra problem: you need to make the call: which one of these are going to be 😛

svgeesus commented:
> However: @svgeesus: if you do accept this wildargs thingy, but you would rather prefer to take it under a different license (e.g. pngcheck's own license), please tell me so, and I'll be happy to relicense.

This is related to

 - https://github.com/pnggroup/pngcheck/issues/36

I'm fine to change the pngcheck license, especially if that helps adoption. I want to see the new version being shipped with Linux distros, for example. The main things in the way of changing license are:

- which one and why
- getting agreement from potentially lots of people for the change

svgeesus commented:
Overall I think I prefer this one to [the alternative](https://github.com/pnggroup/pngcheck/pull/45).

ctruta commented:
> Overall I think I prefer this one to [the alternative](https://github.com/pnggroup/pngcheck/pull/45).

Both solutions have pros and cons, and I, too, agree that this solution is the preferable one, in spite of the other being much smaller. This one is easier to handle at the build and packaging level (as long as "nobody" looks inside `wildargs.c`), and also, easier to retrofit to other build systems, including makefiles, if needed.

> Having all this build documentation as a comment inside the source code doesn't add anything except a maintenance headache as it can diverge from or contradict the actual build scripts and instructions.

I agree. I suppose this is a relic from the distant past when the entire software distribution used to comprise a single file, easy to exchange over floppy discs and emails and whatnot, without breaking apart.

> My preference would be to check that there is nothing useful remaining in there, then delete the build-related parts of the comments.

Will do.

ctruta commented:
> > My preference would be to check that there is nothing useful remaining in there, then delete the build-related parts of the comments.
> 
> Will do.

Done. Mind you, there are still comments that gotta go, and I want to address those ones in a separate commit dedicated strictly to code cleanup. In this commit, I just deleted the whole comment section in which MSVC and `setargv.obj` were mentioned because they happened to be in direct contradiction with what I'm trying to achieve.

I also rewrote the commit message. The new one is even longer than the old one, but at least one paragraph is rewritten in point form and (IMO) it's easier to read.

ace-dent commented:
@ctruta 

Sorry, late to the party. Is this a build time option ? 
I'm keen to compile very lean binaries and have no need for wildcard processing.
